src/components/medication-dispense-review.component.tsx

```diff
@@ -14,30 +14,135 @@ import {
   useMedicationCodeableConcept,
   useMedicationFormulations,
 } from "../medication/medication.resource";
+import { useMedicationRequest } from "../medication-request/medication-request.resource";
+import { PharmacyConfig } from "../config-schema";
+import {
+  useOrderConfig,
+  useSubstitutionReasonValueSet,
+  useSubstitutionTypeValueSet,
+} from "../medication-dispense/medication-dispense.resource";
 
 interface MedicationDispenseReviewProps {
   medicationDispense: MedicationDispense;
   index: number;
   updateMedicationDispense: Function;
-  drugDosingUnits: Array<CommonConfigProps>;
-  drugDispensingUnits: Array<CommonConfigProps>;
-  drugRoutes: Array<CommonConfigProps>;
-  orderFrequencies: Array<CommonConfigProps>;
 }
 
 const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
   medicationDispense,
   index,
   updateMedicationDispense,
-  drugDosingUnits,
-  drugDispensingUnits,
-  drugRoutes,
-  orderFrequencies,
 }) => {
   const { t } = useTranslation();
+  const config = useConfig() as PharmacyConfig;
   const [isEditingFormulation, setIsEditingFormulation] = useState(false);
+  const [isSubstitution, setIsSubstitution] = useState(false);
+  // Dosing Unit eg Tablets
+  const [drugDosingUnits, setDrugDosingUnits] = useState([]);
+  // Dispensing Unit eg Tablets
+  const [drugDispensingUnits, setDrugDispensingUnits] = useState([]);
+  // Route eg Oral
+  const [drugRoutes, setDrugRoutes] = useState([]);
+  // Frequency eg Twice daily
+  const [orderFrequencies, setOrderFrequencies] = useState([]);
```

Comment:
When adding the new substitution options for the two new substitution dropdowns, I realized that we could pull all the config stuff down into the component... I think SWR caching will make sure it doesn't re-fetch the options each time?  

This saves passing all this stuff down as props from the dispense-form component, but let me know if I've got something wrong @pirupius

Reply:
You have a point about the caching. The main worry initially was to avoid making the repeated calls.

Reply:
Yeah, I *think* with SWR it handles caching for you, but we can refactor back if it doesn't.

It also got me thinking about other places where we *don't* want it to cache... but I've got a separate to-do to myself to ticket that